docs/develop/go/message-passing.mdx

```diff
@@ -345,13 +351,76 @@ updateHandle, err := temporalClient.UpdateWorkflow(context.Background(), client.
 if err != nil {
     log.Fatalf("Unable to update workflow: %v", err)
 }
+
 var previousLang message.Language
-err = updateHandle.Get(context.Background(), &previousLang)
+err = updateHandle.Get(ctxWithTimeout, &previousLang)
 if err != nil {
     log.Fatalf("Unable to get update result: %v", err)
 }
 ```
 
+#### Update-With-Start {#update-with-start}
+
+Update-With-Start lets you send an Update that checks whether an already-running Workflow with that ID exists.
+If it does, the Update is delivered.
+If not, it starts a new Workflow Execution with the supplied ID.
+Upon starting a new Workflow, it immediately processes the Update.
+
+To create a new Update operation, use [`client.NewUpdateWithStartWorkflowOperation`](https://pkg.go.dev/go.temporal.io/sdk/client#NewUpdateWithStartWorkflowOperation).
+The Workflow Id is optional - but when it is specified, it must match the one in `StartWorkflowOptions`.
+It is invalid to set the Run ID since there might not be running Workflow Execution.
+
+Use the `WithStartOperation` option to attach an Update Operation to the [`Client.ExecuteWorkflow`](https://pkg.go.dev/go.temporal.io/sdk/client#Client.ExecuteWorkflow) API call.
+Not all `ExecuteWorkflow` options are allowed with Update-With-Start.
+For example, specifying a Cron Schedule results in an error.
+Refer to the API documentation for further details.
```

Comment:
Permalink?

Reply:
Yes please

Reply:
I thought users could get there from the link that's there, but it does take 2 clicks -  so I added a direct link 👍